### PR TITLE
Fix: --fixtures only shows fixtures from first file

### LINKED ISSUE #833: --fixtures only shows fixtures from first file
If I have two files:

`test_a.py`:

```python
import pytest

@pytest.fixture
def fix_a():
    """Fixture A"""
    pass

def test_a(fix_a):
    pass
```

and `test_b.py`:

```python
import pytest

@pytest.fixture
def fix_b():
    """Fixture B"""
    pass

def test_b(fix_b):
    pass
```

Running `pytest --fixtures` will only show `fix_a`:

```
====================== test session starts ======================
platform linux -- Python 3.4.3 -- py-1.4.30 -- pytest-2.7.2
rootdir: /home/florian/tmp/fixture_test, inifile: 
plugins: mock
collected 2 items 
[...]

----------------- fixtures defined from test_a ------------------
fix_a
    Fixture A

=======================  in 0.01 seconds ========================
```

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #833

### CODE REVIEW

**Comment by RonnyPfannschmidt** [comment]
well done

**Comment by nicoddemus** [comment]
Thanks. devpi.net seems to be down though...

**Comment by hpk42** [comment]
devpi is back up -- how can one restart travis?

**Comment by The-Compiler** [comment]
There's a restart button [on the build page](https://travis-ci.org/pytest-dev/pytest/builds/70648629). I just triggered a rebuild.

File: CHANGELOG [hunk 1]
```diff
@@ -4,6 +4,11 @@
 - preserve warning functions after call to pytest.deprecated_call. Thanks
   Pieter Mulder for PR.
 
+- fix issue833: --fixtures does not show fixtures defined in test files,
+  except for the first one collected. Thanks Florian Bruhin for reporting
+  and Bruno Oliveira for the PR.
```
Comment (hpk42) [inline CHANGELOG]: i find the changelog entry a bit confusing.  it should rather read "--fixtures now shows all fixtures of collected test files, not just the first one" or so.

Comment (nicoddemus) [inline CHANGELOG]: Agreed, done!

**Pull request merged**